### PR TITLE
Safer tmpname

### PR DESCRIPTION
**Pull request by mathstuf**
---

This is only lightly tested so far.

### CODE REVIEW

**Comment by rohieb** [comment]
Ah okay, my window managers didn't have problems with disappearing images so far. However, I would very much avoid cluttering the file system with temp files, so I think it's best to choose a random temporary filename for the runtime of the script, re-use it across notifications and then delete it at the end. Unfortunately I cannot test it with my current mpv, which keeps segfaulting for unknown reasons…

**Comment by mathstuf** [comment]
It seems that dunst only keeps a reference to the file, so if the file changes, the notification does as well. I'll file a dunst bug to see if it caching what it shows is more feasible (which sounds like a better idea).

**Comment by mathstuf** [comment]
The first commit here is useful no matter what dunst does. Want a separate PR?

**Comment by rohieb** [comment]
Merged, thanks! Requiring posix really makes sense here, but I did not know that module back then :)

File: notify.lua [hunk 1]
```diff
@@ -71,14 +71,17 @@ http = require("socket.http")
...
+
 local CACHE_DIR = os.getenv("XDG_CACHE_HOME")
 CACHE_DIR = CACHE_DIR or os.getenv("HOME").."/.cache"
 CACHE_DIR = CACHE_DIR.."/mpv/coverart"
 print_debug("making " .. CACHE_DIR)
 os.execute("mkdir -p -- " .. string.shellescape(CACHE_DIR))
 
 function tmpname()
-	return "/tmp/mpv-coverart." .. math.random(0,0xffffff)
+	fd, fname = posix.mkstemp(CACHE_DIR .. "/rescale.XXXXXX")
```
Comment (mathstuf) [inline notify.lua:83]: Hrm, looking at this again. I wonder if this doesn't leak open files :/ .

Comment (rohieb) [inline notify.lua:83]: Oh, good point, there should probably be a `local` in front.

Comment (mathstuf) [inline notify.lua:83]: `fd` is just an integer I think. I don't know Lua that well though. It might need an `os.close` or something.

**Pull request closed without merging**